File: algorithm/001-two_sum/csharp/solution.cs
using System.Collections.Generic;
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
      Dictionary<int, int> complements = new Dictionary<int, int>();
      for (int i = 0; i < nums.Length; i++) {
        int c = target - nums[i];
        if (complements.ContainsKey(c)) {
          return new int[]{complements[c], i};
        } else {
          complements[nums[i]] = i;
        }
      }
      return null;
    }
}

File: algorithm/001-two_sum/scala/solution.scala
import scala.collection.immutable.HashMap

object Solution {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    var complements = HashMap.empty[Int, Int]
    var i = 0
    while (!complements.contains(target - nums(i))) {
      complements += (nums(i) -> i)
      i += 1
    }

    Array(complements(i), i)
  }
}